jwst/tso_photometry/tso_photometry_step.py

```diff
@@ -24,20 +24,19 @@ class TSOPhotometryStep(Step):
 
     def process(self, input):
         with CubeModel(input) as model:
-            # TODO:  need information about the actual source position in
-            # TSO imaging mode (for all subarrays).
-            # Meanwhile, this is a placeholder representing the geometric
-            # center of the image.
-            nint, ny, nx = model.data.shape
-            xcenter = (ny - 1) / 2.
-            ycenter = (ny - 1) / 2.
+            xcenter = model.meta.wcsinfo.crpix1 - 1    # 1-based origin
+            ycenter = model.meta.wcsinfo.crpix2 - 1    # 1-based origin
 
+            # these aperture definitions are for NIRCam
+            # TODO:  need MIRI aperture definitions
             # all radii are in pixel units
             if model.meta.instrument.pupil == 'WLP8':
+                # weak-lens exposure
                 radius = 50
```

Comment:
Be aware that SUB64 (64x64 image) is an allowed subarray size for NIRCam TSImage and the APT does NOT stop the user from selecting the WLP8 with that subarray, in which case these radii are going to fall outside the limits of the image. The TSO WG is currently coming to consensus on what should be done in that case (e.g. use the entire image, with no background region, use a much smaller aperture that only covers the PSF core, ...)